internal/handlers/metric_handler.go

```diff
@@ -22,21 +23,41 @@ var allowedTypes = map[string]string{
 	"counter": "counter",
 }
 
-func SaveMetrics(s repositories.MetricSaver) http.HandlerFunc {
+type ResultError struct {
+	Error string
+}
+
+func SendErrorWithString(w http.ResponseWriter, stringVal string) error {
+	err := json.NewEncoder(w).Encode(
+		&ResultError{
+			stringVal,
+		})
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
можно проще или вообще сразу возвращать результат без переменной. Хотя может в таком случае вообще большого смысла в отдельной функции нет. Хотя как обертка норм.
```suggestion
	return err
```

Reply:
переделал